### PR TITLE
Makefile improvements

### PR DESCRIPTION
**Pull request by Hedingber**
* Enabling to push specific images (e.g. `make push-api push-base`)
* separating legacy images to their own rules
* renaming-s

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -40,35 +40,35 @@ build: docker-images package-wheel ## Build all artifacts
...
 	mlrun
 
-docker-images: $(DOCKER_IMAGES_RULES) ## Build all docker images
+docker-images: $(DEFAULT_DOCKER_IMAGES_RULES) ## Build all docker images
 	@echo Done.
 
 
 push-docker-images: docker-images ## Push all docker images
-	@for image in $(IMAGES_TO_PUSH); do \
+	@for image in $(DEFAULT_IMAGES); do \
```
Comment (liranbg) [inline Makefile:55]: I will suggest again `echo $(DEFAULT_IMAGES) | xargs docker push`, this is IO bound, and parallelizing it will boost up your pushing speed

Comment (tebeka) [reply Makefile:55]: The [documentation](https://docs.docker.com/engine/reference/commandline/push/#concurrent-uploads) says: 

> By default the Docker daemon will push five layers of an image at a time.

I'm not sure how using `xargs -P` will improve the upload time. Is upload time an issue?

Comment (Hedingber) [reply Makefile:55]: @tebeka @liranbg
tl;dr - I'm going to parallelize that
1. Yes, the upload time is an issue, our images are HUGE, we definitely can optimize and decrease their size, but the pushing speed is also part of the game.
2. Indeed docker daemon has the `--max-concurrent-uploads` property which is 5 by default, meaning that regardless of how many parallel processes we'll run, no more than 5 layers will be pushed concurrently. BUT, I tested it (pushing `api`, `base` and `models` images) and with parallel execution it took 29 minutes while the sequential push took 1 hour.
Why ? Because in the sequential push mostly what happens is that the smaller layers pushed first fast, but then most of the time wasting on pushing the big layer, in those moments it doesn't push 5 layers concurrently but 1/2, with the parallel execution we're always using the max possible concurrent uploads... so I'm going to take Liran's suggestion and parallelize that
